app/controllers/api/v1/twilio_controller.rb

```diff
@@ -39,10 +39,20 @@ def establish_ride_zone
     def establish_user_conversation
       to_phone = PhonyRails.normalize_number(params['To'])
       from_phone = PhonyRails.normalize_number(params['From'])
-      @user = User.find_by_phone_number_normalized(from_phone)
-      if @user.nil?
-        @user = User.create!(name: '', user_type: :voter, ride_zone: @ride_zone, password: '12345678', phone_number: from_phone, phone_number_normalized: from_phone, email: "#{from_phone}@example.com")
-        # todo: user role as passenger?
+      sms_name = User.sms_name(from_phone)
+      attrs = {
+          name: sms_name,
+          user_type: :voter,
+          password: '12345678',
+          phone_number: from_phone,
+          phone_number_normalized: from_phone,
+          email: "#{Time.now.to_f}@example.com"
+      }
+      # create the user and rescue duplicate record error to find existing one
+      @user = begin
+        User.create!(attrs)
+      rescue ActiveRecord::RecordNotUnique
```

Comment:
In Rails, the convention is to add a (validates_uniqueness_of)[http://api.rubyonrails.org/classes/ActiveRecord/Validations/ClassMethods.html#method-i-validates_uniqueness_of] validation.  Relying on DB validations is discouraged.

Although, in this case, I think you could do a [https://github.com/DevProgress/drive.vote/pull/45](find_or_create_by) anyway.  Scope it to the email/phone, and create a new one if nothing comes back.